docs/pages/api/rbac.mdx

```diff
@@ -0,0 +1,1017 @@
+		context.Background(),
+		&role,
+	); err != nil {
+		return trace.Wrap(err)
+	}
+	fmt.Println("Upserted Teleport role:", role.GetName())
+	return nil
+}
+```
+
+While the overall behavior of this function is the same as
+`createTeleportRoleFromClusterRoleBinding`, Kubernetes role bindings require
+some differences in how we assign fields to Teleport roles:
+
+- When setting the name of the role, we use the role binding's namespace as the
+  suffix, rather than `-cluster`, to indicate the namespace that this role
+  applies to.
+- In the role's `kubernetes_resources` field, the value has the same namespace
+  as the role binding, rather than applying to all namespaces.
+
+### Create Teleport roles based on Kubernetes resources
+
+Now that we have functions to create Teleport roles based on individual
+Kubernetes RBAC resources, we can fetch all RBAC resources from our Kubernetes
+cluster and call these functions.
+
+Add the following to `main.go`:
+
+```go
+func createTeleportRolesForKubeCluster(teleport *client.Client, k types.KubeCluster) error {
+	rbac, err := getRBACClient()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	crb, err := rbac.ClusterRoleBindings().List(
+		context.Background(),
+		metav1.ListOptions{},
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for _, i := range crb.Items {
+		if err := createTeleportRoleFromClusterRoleBinding(teleport, k, i); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	rb, err := rbac.RoleBindings("").List(
+		context.Background(),
+		metav1.ListOptions{},
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for _, i := range rb.Items {
+		if err := createTeleportRoleFromRoleBinding(teleport, k, i); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+```
+
+`createTeleportRolesForKubeCluster` takes a Teleport client and a
+Teleport-registered Kubernetes cluster. It calls the `getRBACClient` function we
+defined earlier to set up a client for the Kubernetes cluster. It then:
+
+- Lists Kubernetes cluster role bindings and creates a Teleport role for each one.
+- Lists Kubernetes role bindings and creates a Teleport role for each one. 
+
+### Initialize clients and start the application
+
+The functions we declared earlier require a Teleport API client and a
+Teleport-registered Kubernetes cluster, and we will initialize these in the
+entrypoint of the program, the `main` function.
+
+Add the following to `main.go`:
+
+```go
+func main() {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, initTimeout)
+	defer cancel()
+	creds := client.LoadIdentityFile(identityFilePath)
+
+	teleport, err := client.New(ctx, client.Config{
+		Addrs:       []string{proxyAddr},
+		Credentials: []client.Credentials{creds},
+		DialOpts: []grpc.DialOption{
+			grpc.WithReturnConnectionError(),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Connected to Teleport")
+
+	ks, err := teleport.GetKubernetesServers(context.Background())
```

Comment:
Are you intentionally using the background context here instead of the one you created above?

Reply:
I wanted to use a separate context for the initial Teleport connection without the same timeout